barbican/cmd/barbican_manage.py

```diff
@@ -344,10 +347,38 @@ def _verify_label_does_not_exist(self, key_type, label, session):
             print("The label {label} already exists!".format(label=label))
             sys.exit(1)
 
+class SAPCommands(object):
+
+    description = "Move all secrets associated with the old_project_id to the new_project_id."
+
+    @args('--db-url', '-d', metavar='<db-url>', dest='dburl',
+          help='Barbican database URL')
+    @args('--old-project-id', '-o', metavar='<old-project-id>', dest='old_project_id',
+          help='The old project ID to move secrets from.')
+    @args('--new-project-id', '-n', metavar='<new-project-id>', dest='new_project_id',
+          help='The new project ID to move secrets to.')
+    @args('--verbose', '-V', action='store_true', dest='verbose',
+          default=False, help='Show full information about the secret movement.')
+    def move_secrets(self, conf, dburl=None, old_project_id=None, new_project_id=None, verbose=None):
+        if dburl is None:
+            dburl = conf.sql_connection
+
+        engine = create_engine(dburl)
+        with engine.connect() as connection:
```

Comment:
Optional: I don't like that we're not doing the standard barbican thing here. Looking e.g. at the clean command, it sets up the DB like so:
```
        if sql_url:
            CONF.set_override('sql_connection', sql_url)
        repo.setup_database_engine_and_factory()
```
then goes on with
```
    session = repo.get_session()
    …
    query = session.query(models.Project)
    …
    repo.commit()
```

On the last line: Does the connection we create here commit automatically?